tests/Unit/ControlSystem/Systems/Test_Shape.cpp

```diff
@@ -0,0 +1,513 @@
+    initial_ah_coefs[1][iter.set(0, 0)()] = 0.0;
+    initial_ah_coefs[2][iter.set(0, 0)()] = 0.0;
+
+    // Initialize the expected function of time
+    const double dt = 0.01;
+    double time = initial_time + dt;
+    expected_ah_coefs = {initial_time, initial_ah_coefs, time};
+
+    // Update it's derivative often so the function is smooth
+    DataVector updated_deriv = zero_dv;
+
+    const auto update_deriv = [&updated_deriv, &iter, &time, &amplitude, &freqs,
+                               &offsets]() {
+      for (iter.reset(); iter; ++iter) {
+        if (iter.l() == 0 or iter.l() == 1) {
+          continue;
+        }
+        const double freq = freqs[iter()];
+        const double offset = offsets[iter()];
+        if (DerivOrder == 2) {
+          updated_deriv[iter()] =
+              -amplitude * square(freq) * sin(freq * time + offset);
+        } else {
+          updated_deriv[iter()] =
+              -amplitude * cube(freq) * cos(freq * time + offset);
+        }
+      }
+    };
+
+    while (time < final_time) {
+      update_deriv();
+      expected_ah_coefs.update(time, updated_deriv, time + dt);
+      time += dt;
+    }
+
+    // Use func_eps for both because the function is changing so rapidly
+    test_shape_control(generator, make_not_null(&system_helper), initial_time,
+                       final_time, l_max, expected_ah_coefs, func_eps,
+                       func_eps);
+  }
+}
+
+void test_names() {
+  using shape = control_system::Systems::Shape<::ah::ObjectLabel::A, 2>;
+
+  CHECK(pretty_type::name<shape>() == "ShapeA");
+
+  const size_t l_max = 4;
+  SpherepackIterator iter(l_max, l_max);
+  const size_t size = iter.spherepack_array_size();
+
+  // Check known valid indices
+  for (iter.reset(); iter; ++iter) {
+    const auto component_name = shape::component_name(iter(), size);
+    CHECK(component_name.has_value());
+    const std::string check_name =
+        "(l,m)=("s + get_output(iter.l()) + ","s + get_output(iter.m()) + ")"s;
+    CHECK(*component_name == check_name);
+  }
+
+  // Check all indices
+  iter.reset();
+  for (size_t i = 0; i < size; i++) {
+    const auto compact_index = iter.compact_index(i);
+    const auto component_name = shape::component_name(i, size);
+    CHECK(compact_index.has_value() == component_name.has_value());
+    if (component_name.has_value()) {
+      CHECK(*compact_index == iter.current_compact_index());
+      const std::string check_name = "(l,m)=("s + get_output(iter.l()) + ","s +
+                                     get_output(iter.m()) + ")"s;
+      CHECK(*component_name == check_name);
+      ++iter;
+    }
+  }
+}
+
+// [[Timeout, 15]]
+SPECTRE_TEST_CASE("Unit.ControlSystem.Systems.Shape", "[ControlSystem][Unit]") {
+  MAKE_GENERATOR(generator);
+  domain::FunctionsOfTime::register_derived_with_charm();
+
+  test_names();
+
+  // For some of the faster changing AHs, (quadratic and sinusoid) the control
```

Comment:
What counts as fast?  I wouldn't usually consider a sinusoid with a period of at least 150 to be rapidly varying, although I'm not sure what the relevant timescale to compare to is.

Reply:
Not necessarily faster, but I noticed the constant offset for the quadratic and sinusoidal coefficients of the AHs, and I've seen this behavior in SpEC sometimes.